src/rdb_protocol/order_util.cc

```diff
@@ -33,7 +33,7 @@ build_comparison(const term_t* target,
 
     counted_t<const func_t> comparison_function = arg->as_func(GET_FIELD_SHORTCUT);
     rcheck_target(target,
-                  comparison_function->is_deterministic() != deterministic_t::no,
+                  (comparison_function->is_deterministic() & ~deterministic_t::SINGLE_SERVER) == deterministic_t::DETERMINISTIC,
```

Comment:
Why does this not permit r.now() to be constant?  (I have no idea whether it should.)